File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/sale/model/entity/SaleBillingEntity.java
package com.ccadmin.app.sale.model.entity;

import com.ccadmin.app.person.model.entity.PersonEntity;
import com.ccadmin.app.shared.model.entity.AuditTableEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;

import java.io.Serializable;

@Entity
@Table(name = "sale_billing")
public class SaleBillingEntity extends AuditTableEntity implements Serializable {

    @Id
    public String SaleCod;
    public String PersonCod;
    public String DocumentTypeRequest;
    public String DocumentType;
    public String DocumentNum;
    public String LegalName;
    public String CommercialName;
    public String Address;
    public String UbigeoCod;

    @Transient
    public PersonEntity Person;
}

File: frontend/web_ccadmin_lt/src/app/enterprise/sale/model/entity/SaleBillingEntity.ts
import { PersonEntity } from "src/app/enterprise/person/model/entity/PersonEntity";
import { AuditTableEntity } from "src/app/enterprise/shared/model/entity/AuditTableEntity";

export class SaleBillingEntity extends AuditTableEntity {
    public SaleCod: string = "";
    public PersonCod: string = "";
    public DocumentTypeRequest: string = "";
    public DocumentType: string = "";
    public DocumentNum: string = "";
    public LegalName: string = "";
    public CommercialName: string = "";
    public Address: string = "";
    public UbigeoCod: string = "";
    public Person: PersonEntity = new PersonEntity();
}
